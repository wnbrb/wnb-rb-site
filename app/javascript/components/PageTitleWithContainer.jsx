import React from 'react';
import PropTypes from 'prop-types';
import PageTitle from 'components/PageTitle';

const PageTitleWithContainer = ({ text }) => {
    return (
        <div className="max-w-[73rem] px-10 md:px-0 mx-auto my-10 sm:my-20">
            <PageTitle text={text} />
        </div>
    );
};

PageTitleWithContainer.propTypes = {
    text: PropTypes.string,
};

export default PageTitleWithContainer;
