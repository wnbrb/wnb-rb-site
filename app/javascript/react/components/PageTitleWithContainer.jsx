import React from 'react';
import PropTypes from 'prop-types';
import PageTitle from './PageTitle';

const PageTitleWithContainer = ({ text }) => {
    return (
        <div className="max-w-[73rem] px-10 lg:px-8 xl:px-0 mx-auto my-10 sm:my-20">
            <PageTitle text={text} />
        </div>
    );
};

PageTitleWithContainer.propTypes = {
    text: PropTypes.string,
};

export default PageTitleWithContainer;
