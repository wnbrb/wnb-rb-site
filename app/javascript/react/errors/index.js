export class UnauthorizedError extends Error {
    constructor(message) {
        super(message);
        this.name = 'UnauthorizedError';
    }
}
