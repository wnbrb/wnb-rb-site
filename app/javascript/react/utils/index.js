const MS_IN_DAY = 24 * 60 * 60 * 1000;

export const postedAtString = (createdAtString) => {
    const today = new Date();
    const postedDate = new Date(createdAtString);

    const diff = (+today - +postedDate) / MS_IN_DAY;

    if (diff <= 1) {
        return 'today';
    }

    if (diff < 7) {
        return `${Math.floor(diff)} days ago`;
    }

    if (diff < 30) {
        const weekDiff = Math.floor(diff / 7);
        const weekOrWeeks = weekDiff === 1 ? 'week' : 'weeks';

        return `${weekDiff} ${weekOrWeeks} ago`;
    }

    const monthDiff = Math.floor(diff / 30);
    const monthOrMonths = monthDiff === 1 ? 'month' : 'months';

    return `${monthDiff} ${monthOrMonths} ago`;
};
