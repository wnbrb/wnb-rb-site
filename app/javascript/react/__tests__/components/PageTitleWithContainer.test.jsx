import React from 'react';
import { render } from '@testing-library/react';
import PageTitleWithContainer from '../../components/PageTitleWithContainer';

// Mock the PageTitle component to avoid external dependencies in snapshot
jest.mock('../../components/PageTitle', () => {
    // eslint-disable-next-line react/prop-types
    function MockPageTitle({ text }) {
        return <div className="mock-page-title">Mock PageTitle: {text}</div>;
    }
    return MockPageTitle;
});

describe('PageTitleWithContainer Component', () => {
    test('renders page title with container with default props', () => {
        const { container } = render(<PageTitleWithContainer />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders page title with container and text prop', () => {
        const { container } = render(<PageTitleWithContainer text="Welcome" />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders page title with container and longer text', () => {
        const { container } = render(<PageTitleWithContainer text="This is a longer page title" />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('applies correct container CSS classes', () => {
        const { container } = render(<PageTitleWithContainer text="Test" />);

        const containerDiv = container.firstChild;
        expect(containerDiv).toHaveClass('max-w-[73rem]');
        expect(containerDiv).toHaveClass('px-10');
        expect(containerDiv).toHaveClass('lg:px-8');
        expect(containerDiv).toHaveClass('xl:px-0');
        expect(containerDiv).toHaveClass('mx-auto');
        expect(containerDiv).toHaveClass('my-10');
        expect(containerDiv).toHaveClass('sm:my-20');
    });

    test('passes text prop correctly to PageTitle component', () => {
        const testText = 'My Page Title';
        const { getByText } = render(<PageTitleWithContainer text={testText} />);

        expect(getByText(`Mock PageTitle: ${testText}`)).toBeInTheDocument();
    });

    test('renders PageTitle component inside container', () => {
        const { container } = render(<PageTitleWithContainer text="Test" />);

        const mockPageTitle = container.querySelector('.mock-page-title');
        expect(mockPageTitle).toBeInTheDocument();
    });
});
