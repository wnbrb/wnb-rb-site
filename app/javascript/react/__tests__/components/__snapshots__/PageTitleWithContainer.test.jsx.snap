// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitleWithContainer Component renders page title with container and longer text 1`] = `
<div
  class="max-w-[73rem] px-10 lg:px-8 xl:px-0 mx-auto my-10 sm:my-20"
>
  <div
    class="mock-page-title"
  >
    Mock PageTitle: 
    This is a longer page title
  </div>
</div>
`;

exports[`PageTitleWithContainer Component renders page title with container and text prop 1`] = `
<div
  class="max-w-[73rem] px-10 lg:px-8 xl:px-0 mx-auto my-10 sm:my-20"
>
  <div
    class="mock-page-title"
  >
    Mock PageTitle: 
    Welcome
  </div>
</div>
`;

exports[`PageTitleWithContainer Component renders page title with container with default props 1`] = `
<div
  class="max-w-[73rem] px-10 lg:px-8 xl:px-0 mx-auto my-10 sm:my-20"
>
  <div
    class="mock-page-title"
  >
    Mock PageTitle: 
  </div>
</div>
`;
